### PR TITLE
Additions #2

### PR DESCRIPTION
**Pull request by zkayns**
- Added more mod descriptions
- Added click interval setting to AutoClick to allow the user to customize the interval at which the player clicks
- Added apostrophe to Comment History Bypass' description
- Added Auto Kill
- Fix radioButton color in Megahack theme

### CODE REVIEW

File: mod.json [hunk 1]
```diff
@@ -13,7 +13,7 @@
 		"source": "https://github.com/EclipseMenu/EclipseMenu"
 	},
    	"tags": ["cheats", "gameplay", "utility", "customization", "interface"],
-	"developers": ["Eclipse Team", "ninXout", "Prevter", "maxnu", "Firee", "SpaghettDev"],
+	"developers": ["Eclipse Team", "ninXout", "Prevter", "maxnu", "Firee", "SpaghettDev", "Uproxide", "AzureBluet"],
```
Comment (FireMario211) [inline mod.json:16]: no

**Review by FireMario211 — CHANGES_REQUESTED** [review]
contributors != developers

**Review by FireMario211 — APPROVED** [review]

File: src/hacks/Global/AudioSpeed.cpp
Comment (Prevter) [inline src/hacks/Global/AudioSpeed.cpp:1]: i don't like this, it shouldn't override existing settings

File: src/hacks/Bypass/TestmodeBypass.cpp
Comment (Prevter) [inline src/hacks/Bypass/TestmodeBypass.cpp:1]: you broke startpos switcher with this 💔

**Review by Prevter — CHANGES_REQUESTED** [review]

File: src/hacks/Player/AutoKill.cpp
[the remark this answers is not in the document]
Comment (zkayns) [reply src/hacks/Player/AutoKill.cpp:1]: fixed

File: src/hacks/Global/AudioSpeed.cpp
Comment (zkayns) [reply src/hacks/Global/AudioSpeed.cpp:1]: removed because its mid

File: src/hacks/Bypass/TestmodeBypass.cpp
Comment (zkayns) [reply src/hacks/Bypass/TestmodeBypass.cpp:1]: im not even going to pretend i knew what i was doing when i made it so im just going to remove it

**Review by SpaghettDev — CHANGES_REQUESTED** [review]

File: src/hacks/Player/AutoKill.cpp [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+        }
+
+        [[nodiscard]] const char* getId() const override { return "Auto Kill"; }
+    };
+
+    REGISTER_HACK(AutoKill)
+
+    class $modify(GJBaseGameLayer) {
+        virtual void update(float p0) {
+            auto *playLayer = PlayLayer::get();
```
Comment (SpaghettDev) [inline src/hacks/Player/AutoKill.cpp:27]: you never check if playLayer isnt nullptr (its the case when playtesting in the editor)

**Review by SpaghettDev — CHANGES_REQUESTED** [review]

File: src/hacks/Player/AutoKill.cpp [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+namespace eclipse::hacks::Player {
+
+    class AutoKill : public hack::Hack {
+        void init() override {
+            auto tab = gui::MenuTab::find("Player");
+
+            config::setIfEmpty("player.autockill.toggle", false);
+            config::setIfEmpty("player.autokill", 50.f);
+
+            tab->addFloatToggle("Auto Kill", "player.autokill", 0.f, 100.f, "%.2f%");
```
Comment (SpaghettDev) [inline src/hacks/Player/AutoKill.cpp:17]: no toggle for autokill?

File: src/hacks/Player/Autoclicker.cpp [hunk 4]
```diff
@@ -9,7 +9,11 @@ namespace eclipse::hacks::Player {
     class AutoClicker : public hack::Hack {
         void init() override {
             auto tab = gui::MenuTab::find("Player");
-            tab->addToggle("Auto Clicker", "player.autoclick")->handleKeybinds();
```
Comment (SpaghettDev) [inline src/hacks/Player/Autoclicker.cpp:12]: 😨

**Review by SpaghettDev — CHANGES_REQUESTED** [review]

File: src/hacks/Player/AutoKill.cpp (hunk 2 above)
Comment (SpaghettDev) [reply src/hacks/Player/AutoKill.cpp:27]: actually you do, but only after you call a method on it rofl

File: src/hacks/Player/Autoclicker.cpp (hunk 4 above)
Comment (zkayns) [reply src/hacks/Player/Autoclicker.cpp:12]: did i fix it or am i foolish

File: src/hacks/Player/AutoKill.cpp (hunk 2 above)
Comment (zkayns) [reply src/hacks/Player/AutoKill.cpp:27]: oops

File: src/hacks/Player/Autoclicker.cpp (hunk 4 above)
Comment (SpaghettDev) [reply src/hacks/Player/Autoclicker.cpp:12]: you did

File: src/hacks/Player/AutoKill.cpp (hunk 3 above)
Comment (FireMario211) [reply src/hacks/Player/AutoKill.cpp:17]: thats how addFloatToggle works @SpaghettDev

File: src/modules/gui/styles/megahack.hpp [hunk 5]
```diff
@@ -229,6 +229,7 @@ namespace eclipse::gui::imgui {
 
         void visit(RadioButtonComponent* radioButton) override {
             int value = config::get<int>(radioButton->getId(), radioButton->getValue());
+            if (radioButton->getValue()) ImGui::PushStyleColor(ImGuiCol_CheckMark, ImVec4(1, 1, 1, 1));
```
Comment (Prevter) [inline src/modules/gui/styles/megahack.hpp:232]: memory leak 🥰

Comment (SpaghettDev) [reply src/modules/gui/styles/megahack.hpp:232]: its ok prevter the whole menu is a memory leak :)

**Pull request merged**